pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -195,6 +196,10 @@ func (r *ReconcileToolchainStatus) aggregateAndUpdateStatus(reqLogger logr.Logge
 
 		return r.setStatusNotReady(reqLogger, toolchainStatus, fmt.Sprintf("components not ready: %v", unreadyComponents))
 	}
+	err := r.restoredCheck(reqLogger, toolchainStatus)
```

Comment:
I initially thought this would be a good place to do this check, but maybe it'd be better at the beginning of the `setStatusReady` function instead since it's basically tied to that function?

Reply:
good point, i was initially setting the condition type ready's status as true even if sending the notification failed. Because the notification is sent only when the status is back to ready. But @alexeykazakov also commented saying not to set the status to true [here](https://github.com/codeready-toolchain/host-operator/pull/403#discussion_r611270771), so then it makes even more sense to move this within setStatusReady